src/lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper2.ts

```diff
@@ -93,47 +46,50 @@ export const thorchain: Swapper2 = {
     })
   },
 
-  // TODO: getUnsignedTx isn't consumed anywhere yet. When it is, move the HOF to the caller, so we keep the inner function pure
-  getUnsignedTx: withFromOrXpub(
-    async ({ accountMetadata, tradeQuote, from, xpub, supportsEIP1559 }): Promise<UnsignedTx> => {
-      const { receiveAddress, affiliateBps } = tradeQuote
-      const feeAsset = selectFeeAssetById(store.getState(), tradeQuote.steps[0].sellAsset.assetId)
-      const buyAssetUsdRate = selectUsdRateByAssetId(
-        store.getState(),
-        tradeQuote.steps[0].buyAsset.assetId,
-      )
-      const feeAssetUsdRate = feeAsset
-        ? selectUsdRateByAssetId(store.getState(), feeAsset.assetId)
+  getUnsignedTx: async ({
+    accountMetadata,
+    tradeQuote,
+    from,
+    xpub,
+    supportsEIP1559,
+  }): Promise<UnsignedTx> => {
+    const { receiveAddress, affiliateBps } = tradeQuote
+    const feeAsset = selectFeeAssetById(store.getState(), tradeQuote.steps[0].sellAsset.assetId)
+    const buyAssetUsdRate = selectUsdRateByAssetId(
+      store.getState(),
+      tradeQuote.steps[0].buyAsset.assetId,
+    )
+    const feeAssetUsdRate = feeAsset
+      ? selectUsdRateByAssetId(store.getState(), feeAsset.assetId)
+      : undefined
+
+    if (!buyAssetUsdRate) throw Error('missing buy asset usd rate')
+    if (!feeAssetUsdRate) throw Error('missing fee asset usd rate')
+
+    const accountType = accountMetadata?.accountType
+
+    const chainSpecific =
+      accountType && xpub
+        ? {
+            xpub,
+            accountType,
+            satoshiPerByte: (tradeQuote as TradeQuote<ThorUtxoSupportedChainId>).steps[0].feeData
+              .chainSpecific.satsPerByte,
+          }
         : undefined
 
-      if (!buyAssetUsdRate) throw Error('missing buy asset usd rate')
-      if (!feeAssetUsdRate) throw Error('missing fee asset usd rate')
-
-      const accountType = accountMetadata?.accountType
-
-      const chainSpecific =
-        accountType && xpub
-          ? {
-              xpub,
-              accountType,
-              satoshiPerByte: (tradeQuote as TradeQuote<ThorUtxoSupportedChainId>).steps[0].feeData
-                .chainSpecific.satsPerByte,
-            }
-          : undefined
-
-      const fromOrXpub = from ? { from } : { xpub: xpub! }
-      return await getSignTxFromQuote({
-        tradeQuote,
-        receiveAddress,
-        affiliateBps,
-        chainSpecific,
-        buyAssetUsdRate,
-        ...fromOrXpub,
-        feeAssetUsdRate,
-        supportsEIP1559,
-      })
-    },
-  ),
+    const fromOrXpub = from !== undefined ? { from } : { xpub }
```

Comment:
`from !== undefined` allows tsc to infer `xpub` is typeof string so we dont need non-null assertion

Reply:
i.e `!from` doesnt work for us in this case

Reply:
Absolutely smort